HPCC-13714 Option for #Webservice(Fields()) to include unspecified fields

afishbeck opened the pull request:
The following syntax:
# WEBSERVICE(FIELDS('field1','field2','*'));

The addition of the name '*' would cause the form to show
first field1 and field2 and then all additional stored inputs
(unless noinput is specified for that stored).

Signed-off-by: Anthony Fishbeck anthony.fishbeck@lexisnexis.com

HPCCSmoketest commented:
Automated Smoketest 
Sha: 0d256d8c831e771d96cfb922212dc97425ec9275
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:83 passed:83 failed:0

HPCC Stop: OK
HPCC Uninstall: OK

afishbeck committed:
HPCC-13714 Option for #Webservice(Fields()) to include unspecified fields

The following syntax:
#WEBSERVICE(FIELDS('field1','field2','*'));

The addition of the name '*' would cause the form to show
first field1 and field2 and then all additional stored inputs
(unless noinput is specified for that stored).

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

HPCCSmoketest commented:
Automated Smoketest 
Sha: 4a06d24388c10c03834db92a62a757e60c0f34cd
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:83 passed:83 failed:0

HPCC Stop: OK
HPCC Uninstall: OK

richardkchapman commented:
@wangkx Please review

wangkx commented on esp/services/ws_ecl/ws_ecl_wuinfo.cpp:
```diff
@@ -209,11 +209,39 @@ bool WsEclWuInfo::getWsResource(const char *name, StringBuffer &out)
+                            int seq = fieldSeq.length() ? atoi(fieldSeq.str()) : 0;
+                            if (seq < 0)
+                                seq = 0;
+                            seq = seq + i + 1;
```
I do not understand the logic here. Does the 'fieldSeq' give the offset from current location i? Can we assume that it returns continuous '1, 2, ..., n'?

afishbeck replied:
No the fieldSeq is being adapted to work a bit differently here.  When #WebService is not in use the fieldSeq is a relative position to all the other fields.  I.e. for managing over time you may use sparse numbering..   100, 150, 200, 300.  So I can easily insert a new entry somewhere.

What I am doing here allows you to put those in your #WebService field list, so I respect that numbering but make sure it fits where it belongs in our list.  Basically things inserted here in the list use fieldSeq only to determine there order relative to each other, but not fields named explicitly in the list.

So WebService(Fields('abc', *, def'));
anything added by '*' uses fieldSeq, but must be inserted between abc and def which do not use fieldSeq.

wangkx replied:
Ok

wangkx commented:
@afishbeck looks fine. Just one question.
